src/equal.c

```diff
@@ -121,8 +121,7 @@ static int dbl_equal_scalar(const double* x, const double* y, bool na_equal) {
   if (na_equal) {
     if (R_IsNA(xi)) return R_IsNA(yj);
     if (R_IsNaN(xi)) return R_IsNaN(yj);
-    if (R_IsNA(yj)) return false;
-    if (R_IsNaN(yj)) return false;
+    if (ISNAN(yj)) return false;
```

Comment:
What’s the difference between ISNAN and isnan (used below)?

Reply:
`isnan()` doesn't return `0` on some systems (Windows). We hit this before with https://github.com/r-lib/vctrs/pull/588 but it really only matters if you are returning the value of `ISNAN()`. For this check `isnan()` or `ISNAN()` would work fine, so I'll change to `isnan()` to be consistent with the rest of the function.

Reply:
Re-run benchmarks after changing `ISNAN()` to `isnan()`. Really no change, perhaps slightly faster now.

``` r
library(vctrs)
x <- rep(1, 1e8)
vec_equal_old <- vctrs2::vec_equal

bench::mark(
  vec_equal(x, x), 
  vec_equal_old(x, x), 
  iterations = 20
)
#> # A tibble: 2 x 6
#>   expression               min   median `itr/sec` mem_alloc `gc/sec`
#>   <bch:expr>          <bch:tm> <bch:tm>     <dbl> <bch:byt>    <dbl>
#> 1 vec_equal(x, x)        207ms    221ms      4.50     381MB    1.50 
#> 2 vec_equal_old(x, x)    347ms    380ms      2.64     381MB    0.660
```

``` r
library(vctrs)
x <- rep(1, 1e8)
vec_equal_old <- vctrs2::vec_equal

bench::mark(
  vec_equal(x, x, na_equal = TRUE), 
  vec_equal_old(x, x, na_equal = TRUE), 
  iterations = 20
)
#> # A tibble: 2 x 6
#>   expression                             min   median `itr/sec` mem_alloc
#>   <bch:expr>                           <bch> <bch:tm>     <dbl> <bch:byt>
#> 1 vec_equal(x, x, na_equal = TRUE)     509ms 522.29ms     1.92      381MB
#> 2 vec_equal_old(x, x, na_equal = TRUE)    1s    1.03s     0.951     381MB
#> # … with 1 more variable: `gc/sec` <dbl>
```